### ISSUE TITLE
Not able to switch to WSL2. Error code: Wsl/Service/CreateVm/HCS/HCS_E_HYPERV_NOT_INSTALLED

### ISSUE THREAD

**Issue by ghoebenreich** [microsoft/WSL#11384]
### Windows Version

Microsoft Windows [Version 10.0.22621.3155]

### WSL Version

2.1.5.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.146.1-2

### Distro Version

Ubuntu 22.04

### Repro Steps

I have WSL1 running for the Linux Ubuntu 22.04 with name ubuntu

Executing` wsl --set-version ubuntu 2` issues the error:
For information on key differences with WSL 2 please visit https://aka.ms/wsl2
Conversion in progress, this may take a few minutes.
WSL2 is not supported with your current machine configuration.
Please enable the "Virtual Machine Platform" optional component and ensure virtualization is enabled in the BIOS.
Enable "Virtual Machine Platform" by running: wsl.exe --install --no-distribution
For information please visit https://aka.ms/enablevirtualization
Error code: Wsl/Service/CreateVm/HCS/HCS_E_HYPERV_NOT_INSTALLED

I enabled the "Virtual Machine Platform". I also disabled it /Reboot/ enabled /reboot. No matter what I do I always get the same error. I checked the BIOS and I have VT-x and all other Virtualization options enabled in the BIOS. 

`wsl.exe --install --no-distribution` : 

`The operation completed successfully.`

I'm running this on: Dell Precision 7560 with CPU 11th Gen Intel(R) Core(TM) i9-11950H @ 2.60GHz / Intel64 Family 6 Model 141 Stepping 1, GenuineIntel. No 
My CPU supports SLAT.

`bcdedit /enum | findstr -i hypervisorlaunchtype`
shows
`hypervisorlaunchtype    Auto`

Not sure what settings I still have to check. I captured the WSL logs
[WslLogs-2024-03-27_11-10-14.zip](https://github.com/microsoft/WSL/files/14771799/WslLogs-2024-03-27_11-10-14.zip)

Would be great if someone can tell me what's still wrong.

### Expected Behavior

` wsl --set-version ubuntu 2` 

should give a positive result and switch ubuntu to WSL version 2.

### Actual Behavior

I enabled the "Virtual Machine Platform". I also disabled it /Reboot/ enabled /reboot. No matter what I do I always get the same error. I checked the BIOS and I have VT-x and all other Virtualization options enabled in the BIOS. 

`wsl.exe --install --no-distribution` : 

`The operation completed successfully.`

I'm running this on: Dell Precision 7560 with CPU 11th Gen Intel(R) Core(TM) i9-11950H @ 2.60GHz / Intel64 Family 6 Model 141 Stepping 1, GenuineIntel. No 
My CPU supports SLAT.

`bcdedit /enum | findstr -i hypervisorlaunchtype`
shows
`hypervisorlaunchtype    Auto`

Not sure what settings I still have to check. I captured the WSL logs
[WslLogs-2024-03-27_11-10-14.zip](https://github.com/microsoft/WSL/files/14771799/WslLogs-2024-03-27_11-10-14.zip)

Would be great if someone can tell me what's still wrong.
I looked at similar issues like [Issue 10332](https://github.com/microsoft/WSL/issues/10332) but it did not solve my issue

**Comment by OneBlue** [comment]
Thank you @ghoebenreich. From the logs, everything looks good so I wonder if you have another hypervisor installed ? Maybe VirtualBox or vmware ?

**Comment by ghoebenreich** [comment]
HI @OneBlue, I previously had virtualbox installed and docker for windows, but I removed bot, I'm running this on bare-metal laptop. How do I check what hypervisor is installed, are there some registry entries to check? I also tried to use the hyper-v installation, but I removed it too (deselected Hyper-V on "Tun Windows features on or off" Tab, since I thought I need it for the WSL2, but according to the documentation only "Virtual Machine Platform" is needed which is installed. Any ideas

**Comment by ghoebenreich** [comment]
@OneBlue I only can set it to `auto `or `off`, when I set the `bcdedit /set hypervisorlaunchtype on` I get 

```
PS C:\Windows\System32> bcdedit /set hypervisorlaunchtype bool on
The integer data is not valid as specified.
Run "bcdedit /?" for command line assistance.
The parameter is incorrect.
```

**Comment by benhillis** [comment]
@ghoebenreich - did you try auto?

**Comment by ghoebenreich** [comment]
@benhillis I had `hypervioslaunchtype `always set to auto. I tried to off and auto it again, no change.

**Comment by ghoebenreich** [comment]
@OneBlue , @benhillis does anybody has some more ideas. Since nothing worked for WSL2, I installed Hyper-V from the windows components. Downloaded Ubuntu 22, but also was not able to start. It tells me a Hyper-V component is not running. Maybe this has something to do with it, that some registry setting or permission setting is blocking something.

**Comment by onomatopellan** [comment]
In an elevated PowerShell if you run `systeminfo` what it says at the last part, the "Hyper-V Requirements:"?

**Comment by ghoebenreich** [comment]
@onomatopellan :

 `Hyper-V Requirements:      A hypervisor has been detected. Features required for Hyper-V will not be displayed.`

**Comment by onomatopellan** [comment]
That's correct. Your Windows session is running under the hypervisor. I guess you already tried everything but do you perhaps have [Memory integrity](https://support.microsoft.com/en-us/windows/device-protection-in-windows-security-afa11526-de57-b1c5-599f-3a4c6a61c5e2) disabled? If so enable it and restart the computer.

Also I'd try to enable Windows Sandbox feature:

![Optional Windows Features dlg](https://github.com/microsoft/WSL/assets/599142/b3a74e0d-3373-4ff7-8725-688e95a49f3a)

If you can run it we can discard any Hyper-v issue and point definitely to WSL2.

**Comment by ghoebenreich** [comment]
@onomatopellan Sorry for the late response. I checked the memory integrity. It tells me that there is an incompatible driver installed, but does not list the driver and refuses to enable it :). Maybe that's one of my core issues. Thanks for the hint. I will try to enable Windows Sandbox and report back.

**Comment by HO-COOH** [comment]
None of the method above works for me, not even Windows Sandbox. And I clearly remembered my docker worked before

![image](https://github.com/microsoft/WSL/assets/42881734/ca38d6ff-7534-4d67-bd41-7d578d2f0d9f)

**Comment by ghoebenreich** [comment]
I was able get it working finally. Not exactly sure how, but I did the following. I went back to the BIOS and unchecked anything which is related to HyperV support. Booted again and verified with the systeminfo that I don't have a HyperV. I worked for sometime without changing that. Had some reboots. After a week of working I switched on one setting after the other again, without reinstalling and Windows features. To my surprise I was able to call wsl --default-version 2 and my ubuntu container got updated to use wsl2. Since that it seems to work:

![image](https://github.com/microsoft/WSL/assets/99474508/be0ffc2a-9c49-445e-a12e-fea01ea3532d)

**Comment by zulis** [comment]
I was getting the same error:
```Error code: Wsl/Service/CreateInstance/CreateVm/HCS/HCS_E_HYPERV_NOT_INSTALLED```
But the issue was that I've disabled Core Isolation -> Memory integrity. After setting it back to On problem was solved.

**Comment by TrumX** [comment]
> I was getting the same error: `Error code: Wsl/Service/CreateInstance/CreateVm/HCS/HCS_E_HYPERV_NOT_INSTALLED` But the issue was that I've disabled Core Isolation -> Memory integrity. After setting it back to On problem was solved.

just wanted to say I had similar errors, disabled core isolation was the problem too.

**Comment by elfatairy** [comment]
> Thank you [@ghoebenreich](https://github.com/ghoebenreich). Let's try one thing: Can you run `bcdedit /set hypervisorlaunchtype on` as elevated, reboot and see if that solves the issue ?

Thank u so much. It did work

**Comment by AdamZeman** [comment]
I have noticed wsl not running when I tried to use Docker Desktop. Remembered I have had recently installed VmWare. I did the following steps:

1. Unchecked any hypervisor (VMWare / Virtual Box / ...) on startup
2. Running `bcdedit /set hypervisorlaunchtype auto` in admin terminal

Thanks @OneBlue

**Comment by VladasZ** [comment]
I was able to fix it by disabling virtualization in BIOS, loading windows, then rebooting and enabling it again.

**Comment by OneBlue** [comment]
Closing since the original issue was resolved

**Comment by ptaczek** [comment]
> I was getting the same error: `Error code: Wsl/Service/CreateInstance/CreateVm/HCS/HCS_E_HYPERV_NOT_INSTALLED` But the issue was that I've disabled Core Isolation -> Memory integrity. After setting it back to On problem was solved.

This solved it for me! Tnx!

**Comment by pak0s** [comment]
Make sure `Hyper-V Host Compute Service` is installed and running. In my case both Hyper-V and WSL were not working. Enabling `Hyper-V Virtual Machine Management` was giving low memory error. Enabling `Hyper-V Host Compute Service` resolved both WSL and enabled `Hyper-V Virtual Machine Management` service.
